src/sentry/api/bases/organization_events.py

```diff
@@ -70,12 +68,4 @@ def get_snuba_query_args_legacy(self, request, organization):
             "filter_keys": _filter.filter_keys,
         }
 
-        # 'legacy' endpoints cannot access transactions dataset.
-        # as they often have assumptions about which columns are returned.
-        dataset = snuba.detect_dataset(snuba_args)
-        if dataset != Dataset.Events:
-            raise OrganizationEventsError(
-                "Invalid query. You cannot reference non-events data in this endpoint."
-            )
-
```

Comment:
What are we doing to make sure they aren't referencing transaction data now?

Reply:
I assume this was returning the transactions dataset before, but now snuba will determine dataset - so are we checking somewhere else that they won't be needing transaction data?

Reply:
Previously queries to this endpoint would end up in `dataset_query()` and have the dataset automatically detected. Now that `transform_aliases_and_query` is pinned to the events dataset this check is redundant.